eldritch/events.py

```diff
@@ -2328,22 +2331,31 @@ def resolve(self, state):
     if isinstance(self.items, values.Value):
       self.items = self.items.value(state)
 
-    self.discarded = []
-    for item in self.items:
-      for attr in item.tokens:
-        item.tokens[attr] = 0
-      if item not in self.character.possessions:
-        continue
-      self.character.possessions.remove(item)
-      item._exhausted = False  # pylint: disable=protected-access
-      if hasattr(item, "_active"):
-        item._active = False  # pylint: disable=protected-access
-      if not self.to_box:
-        getattr(state, item.deck).append(item)
-      self.discarded.append(item)
+    if self.discarded is None:
+      self.discarded = []
+      for item in self.items:
+        for attr in item.tokens:
+          item.tokens[attr] = 0
+        if item not in self.character.possessions:
+          continue
+        self.character.possessions.remove(item)
+        item._exhausted = False  # pylint: disable=protected-access
+        if hasattr(item, "_active"):
```

Comment:
Since we are now explicitly deactivating the items, I think this if block can be removed?